### PR TITLE
Add Rollup & build library

### PR DESCRIPTION
**Pull request by Glavin001**
## Why?

Using class properties is not consumable by Storybook, etc so this package requires transpilation.

See https://github.com/pmndrs/drei/pull/933

### Error Logs

```
ERROR in ./node_modules/manipulator3d/dist/Manipulator3D.js 22:24
Module parse failed: Unexpected token (22:24)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| export class Manipulator3D{
|     // #region MAIN
>     mesh                = null;                 // Rendering of the gizmo
|     data                = null;                 // Data state of the gizmo
|     attachedObject      = null;                 // Reference to selected object
 @ ./node_modules/manipulator3d/dist/index.js 3:0-70 5:0-10:2
 @ ./src/core/TransformControls.tsx
 @ ./src/core/index.ts
 @ ./src/web/index.ts
 @ ./src/index.ts
 @ ./.storybook/stories/useBVH.stories.tsx
 @ ./.storybook/stories sync ^\.(?:(?:^|\/|(?:(?:(?!(?:^|\/)\.).)*?)\/)(?!\.)(?=.)[^/]*?\.stories\.(ts|tsx))$
 @ ./generated-stories-entry.js
 @ multi ./node_modules/@storybook/core-client/dist/esm/globals/polyfills.js ./node_modules/@storybook/core-client/dist/esm/globals/globals.js (webpack)-hot-middleware/client.js?reload=true&quiet=false&noInfo=undefined ./storybook-init-framework-entry.js ./node_modules/@storybook/react/dist/esm/client/preview/config-generated-config-entry.js ./node_modules/@storybook/addon-knobs/dist/preset/addDecorator.js-generated-config-entry.js ./node_modules/@storybook/addon-actions/dist/esm/preset/addDecorator.js-generated-config-entry.js ./node_modules/@storybook/addon-actions/dist/esm/preset/addArgs.js-generated-config-entry.js ./.storybook/preview.ts-generated-config-entry.js ./generated-stories-entry.js

ERROR in ./node_modules/manipulator3d/dist/ManipulatorData.js 33:20
Module parse failed: Unexpected token (33:20)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| export class ManipulatorData{
|     // #region MAIN
>     scaleFactor     = 10;           // Rescale axes based on camera distance, to keep gizmo about the same size
|     minFlipAdjust   = -0.02;      // Dot Angle minimum to flip Gizmo
|     minHitDistance  = 0.1;          // Min Distance to consider touching an axis segment
 @ ./node_modules/manipulator3d/dist/index.js 1:0-72 5:0-10:2 5:0-10:2
 @ ./src/core/TransformControls.tsx
 @ ./src/core/index.ts
 @ ./src/web/index.ts
 @ ./src/index.ts
 @ ./.storybook/stories/useBVH.stories.tsx
 @ ./.storybook/stories sync ^\.(?:(?:^|\/|(?:(?:(?!(?:^|\/)\.).)*?)\/)(?!\.)(?=.)[^/]*?\.stories\.(ts|tsx))$
 @ ./generated-stories-entry.js
 @ multi ./node_modules/@storybook/core-client/dist/esm/globals/polyfills.js ./node_modules/@storybook/core-client/dist/esm/globals/globals.js (webpack)-hot-middleware/client.js?reload=true&quiet=false&noInfo=undefined ./storybook-init-framework-entry.js ./node_modules/@storybook/react/dist/esm/client/preview/config-generated-config-entry.js ./node_modules/@storybook/addon-knobs/dist/preset/addDecorator.js-generated-config-entry.js ./node_modules/@storybook/addon-actions/dist/esm/preset/addDecorator.js-generated-config-entry.js ./node_modules/@storybook/addon-actions/dist/esm/preset/addArgs.js-generated-config-entry.js ./.storybook/preview.ts-generated-config-entry.js ./generated-stories-entry.js

ERROR in ./node_modules/manipulator3d/dist/ManipulatorMesh.js 7:18
Module parse failed: Unexpected token (7:18)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| export class ManipulatorMesh extends THREE.Group{
|     // #region MAIN
>     axisColors    = [ 0x81D773, 0x6DA9EA, 0xF7716A ];
|     axisLines     = [];
|     axisArcs      = [];
 @ ./node_modules/manipulator3d/dist/index.js 2:0-72 5:0-10:2
 @ ./src/core/TransformControls.tsx
 @ ./src/core/index.ts
 @ ./src/web/index.ts
 @ ./src/index.ts
 @ ./.storybook/stories/useBVH.stories.tsx
 @ ./.storybook/stories sync ^\.(?:(?:^|\/|(?:(?:(?!(?:^|\/)\.).)*?)\/)(?!\.)(?=.)[^/]*?\.stories\.(ts|tsx))$
 @ ./generated-stories-entry.js
 @ multi ./node_modules/@storybook/core-client/dist/esm/globals/polyfills.js ./node_modules/@storybook/core-client/dist/esm/globals/globals.js (webpack)-hot-middleware/client.js?reload=true&quiet=false&noInfo=undefined ./storybook-init-framework-entry.js ./node_modules/@storybook/react/dist/esm/client/preview/config-generated-config-entry.js ./node_modules/@storybook/addon-knobs/dist/preset/addDecorator.js-generated-config-entry.js ./node_modules/@storybook/addon-actions/dist/esm/preset/addDecorator.js-generated-config-entry.js ./node_modules/@storybook/addon-actions/dist/esm/preset/addArgs.js-generated-config-entry.js ./.storybook/preview.ts-generated-config-entry.js ./generated-stories-entry.js
 ```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,30 +1,45 @@
...
     "dependencies": {
-        "three" : "^0.138.2"
+        "three": "^0.138.2"
     },
-    
     "devDependencies": {
+        "@babel/core": "^7.18.2",
+        "@babel/preset-env": "^7.18.2",
+        "@rollup/plugin-babel": "^5.3.1",
+        "rollup": "^2.75.5",
```
Comment (sketchpunk) [inline package.json:42]: https://vitejs.dev/guide/build.html
Vite comes with rollup already, so there shouldn't be a need for any extra dependencies.

Comment (Glavin001) [reply package.json:42]: Ah thanks! I'm not familiar with Vite yet. Will clean it up and test again.

**Comment by sketchpunk** [comment]
I fixed up the library to build ES and CJS files, plus its now available on NPM without the need to install straight from github.

**Comment by Glavin001** [comment]
Awesome thank you! I'll close this 🎉

**Pull request closed without merging**